usr.sbin/npf/npfctl/npf_parse.y

```diff
@@ -970,6 +987,183 @@ ifref
 	}
 	;
 
+user_id
+	: /* empty */ { $$ = NULL; }
+	| USER uids	{ $$ = $2; }
+	;
+
+uids
+	: uid_item	{ $$ = $1; }
+	;
+
+uid_item
+	: uid
+	{
+		if (($$ = calloc(1, sizeof(struct r_uid))) == NULL)
+			errx(EXIT_FAILURE, "uid_item: calloc");
+		$$->uid[0] = $1;
+		$$->uid[1] = $1;
+		$$->op = NPF_OP_EQ;
+	}
+	| op_unary uid
+	{
+		if (($$ = calloc(1, sizeof(struct r_uid))) == NULL)
+			errx(EXIT_FAILURE, "uid_item: calloc");
+		$$->uid[0] = $2;
+		$$->uid[1] = $2;
+		$$->op = $1;
+	}
+	| uid op_binary uid
+	{
+		if (($$ = calloc(1, sizeof(struct r_uid))) == NULL)
+			errx(EXIT_FAILURE, "uid_item: calloc");
+		$$->uid[0] = $1;
+		$$->uid[1] = $3;
+		$$->op = $2;
+	}
+	;
+
+uid
+	: NUM
+	{
+		if ($1 >= UID_MAX) {
+			yyerror("illegal uid value %lu", $1);
+			YYERROR;
+		}
+		$$ = $1;
+	}
+	| IDENTIFIER
+	{
+		if (($$ = npfctl_parse_user($1)) == 1) {
+			yyerror("unknown user %s", $1);
+			YYERROR;
+		}
+	}
+	| VAR_ID
+	{
+		npfvar_t *vp = npfvar_lookup($1);
+		int type = npfvar_get_type(vp, 0);
+		char *usr;
+
+		switch (type) {
+		case NPFVAR_IDENTIFIER:
+		case NPFVAR_STRING:
+			usr = npfvar_expand_string(vp);
+			if (($$ = npfctl_parse_user(usr)) == 1) {
+				yyerror("unknown user %s", $1);
+				YYERROR;
+			}
+			break;
+		case NPFVAR_NUM:
+			$$ = npfvar_expand_number(vp);
+			break;
+		case -1:
+			yyerror("undefined variable '%s'", $1);
+			break;
+		default:
+			yyerror("wrong variable '%s' type '%s' for user id",
+				$1, npfvar_type(type));
+			break;
+		}
+	}
+	;
+
+group_id
+	: /* empty */ { $$ = NULL; }
+	| USERGROUP gids	{ $$ = $2; }
```

Comment:
Are you calling this "USERGROUP" instead of "GROUP" because group is already used?